lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -168,18 +168,82 @@ class PointerSupportDetector {
       'pointers:$hasPointerEvents, touch:$hasTouchEvents, mouse:$hasMouseEvents';
 }
 
+class _Listener {
+  _Listener._({
+    required this.event,
+    required this.target,
+    required this.handler,
+    required this.useCapture,
+    required this.isNative,
+  });
+
+  /// Registers a listener for the given [event] on [target] using the Dart-to-JS API.
+  factory _Listener.register({
+    required String event,
+    required html.EventTarget target,
+    required html.EventListener handler,
+    bool capture = false,
+  }) {
+    final _Listener listener = _Listener._(
+      event: event,
+      target: target,
+      handler: handler,
+      useCapture: capture,
+      isNative: false,
+    );
+    target.addEventListener(event, handler, capture);
+    return listener;
+  }
+
+  /// Registers a listener for the given [event] on [target] using the native JS API.
+  factory _Listener.registerNative({
+    required String event,
+    required html.EventTarget target,
+    required html.EventListener handler,
+    bool capture = false,
+    bool passive = false,
+  }) {
+    final Object eventOptions = createPlainJsObject(<String, Object?>{
+      'capture': capture,
+      'passive': passive,
+    });
+    final html.EventListener jsHandler = allowInterop((html.Event event) => handler(event));
+    final _Listener listener = _Listener._(
+      event: event,
+      target: target,
+      handler: jsHandler,
+      useCapture: capture,
+      isNative: true,
+    );
+    addJsEventListener(target, event, jsHandler, eventOptions);
+    return listener;
+  }
+
+  final String event;
+
+  final html.EventTarget target;
+  final html.EventListener handler;
+
+  final bool useCapture;
+  final bool isNative;
+
+  void unregister() {
+    if (isNative) {
+      removeJsEventListener(target, event, handler, useCapture);
```

Comment:
I was gonna say:
> Should we pass the same `eventOptions` that were passed to `addJsEventListener`?

But then I realized only the `useCapture` option matters for `removeEventListener`. TIL.

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#matching_event_listeners_for_removal

Reply:
Indeed, I've dug into those when I was doing my research.